_posts/2021-11-22-changelog-160.markdown

```diff
@@ -0,0 +1,110 @@
+
+  {% include gif-img.html path="/assets/posts/changelog-160/resolve_index_from_brackets" w="800px" %}
+
+* [#7826] Add intention to unwrap a (tuple struct or enum variant) constructor from a single expression. Could be useful when you're adding or removing `Option` or `Result` from function signatures during refactoring (by [@Kobzol])
+
+  {% include gif-img.html path="/assets/posts/changelog-160/unwrap_tuple_constructor" w="800px" %}
+
+* [#7799] Support [if let guard](https://rust-lang.github.io/rfcs/2294-if-let-guard.html) syntax
+
+  {% include gif-img.html path="/assets/posts/changelog-160/if_let_guard" w="800px" %}
+
+* [#7725] Take into account all kinds of items generated by [custom derive procedural macros](https://doc.rust-lang.org/reference/procedural-macros.html#derive-macros). Previously only `impl` blocks were accounted. Please, note that procedural macro expansion is experimental. If you want to try it, enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` experimental features
+
+* [#8102] [#8042] Enable [`Change signature` refactoring](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#change-sign) when a type of a trait method parameter changes (by [@Kobzol])
+
+* [#7840] The intention for removing braces from a single expression now also works for blocks that contain a single statement (by [@Kobzol])
+
+* [#6929] Detection of illegal lifetime names (by [@Stzx])
+
+* [#7719] Support the [E0451](https://doc.rust-lang.org/error-index.html#E0451) compiler error ("a struct constructor with private fields was invoked") for struct field literals using [shorthand init](https://doc.rust-lang.org/reference/expressions/struct-expr.html#struct-field-init-shorthand) (by [@Kobzol])
+
+## Performance Improvements
+
+* [#8089] Avoid using potentially long-running computation inside the UI thread when invoking the [`Introduce constant`](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#extractconst-refactoring) refactoring (by [@Kobzol])
+
+* [#8043] Do not use potentially long-running operations on the UI thread in the `Change signature` refactoring (by [@Kobzol])
+
+## Fixes
+
+* Generated sources fixes:
+
+  * [#8012] Improve work with generated sources provided via `org.rust.cargo.evaluate.build.scripts` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features). Now the plugin doesn't pollute `Project Structure` view with entries for generated `out` directories. Also, the custom paths of target directories are also supported
+
+  * [#8056] Fix build script evaluation in [workspace](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html). Note, build script evaluation is disabled by default. To turn it on, enable `org.rust.cargo.evaluate.build.scripts` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features)
```

Comment:
We still write about `org.rust.cargo.evaluate.build.scripts` experimental feature twice. Can we lift this note into `Generated sources fixes` item and write it only once?